src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -16,48 +16,53 @@ internal class ActorProxyFactory : IActorProxyFactory
     {
         private readonly IDaprInteractor daprInteractor;
 
+        /// <inheritdoc/>
+        public ActorProxyOptions DefaultActorProxyOptions { get; set; } = new ActorProxyOptions();
+
         /// <summary>
         /// Initializes a new instance of the <see cref="ActorProxyFactory"/> class.
-        /// TODO: Accept Retry settings.
         /// </summary>
         public ActorProxyFactory()
         {
-            // TODO: Allow configuration of serialization and client settings.
             this.daprInteractor = new DaprHttpInteractor();
         }
 
-        /// <inheritdoc/>
-        public TActorInterface CreateActorProxy<TActorInterface>(ActorId actorId, string actorType)
-            where TActorInterface : IActor
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ActorProxyFactory"/> class.
+        /// </summary>
+        public ActorProxyFactory(ActorProxyOptions options) : this()
         {
-            return (TActorInterface)this.CreateActorProxy(actorId, typeof(TActorInterface), actorType);
+            DefaultActorProxyOptions = options;
         }
 
         /// <inheritdoc/>
-        public ActorProxy Create(ActorId actorId, string actorType)
+        public object CreateActorProxy(ActorId actorId, Type actorInterfaceType, string actorType, ActorProxyOptions options = null)
         {
-            var actorProxy = new ActorProxy();
-            var nonRemotingClient = new ActorNonRemotingClient(this.daprInteractor);
-            actorProxy.Initialize(nonRemotingClient, actorId, actorType);
+            var remotingClient = new ActorRemotingClient(this.daprInteractor);
+            var proxyGenerator = ActorCodeBuilder.GetOrCreateProxyGenerator(actorInterfaceType);
+            var actorProxy = proxyGenerator.CreateActorProxy();
+            actorProxy.Initialize(remotingClient, actorId, actorType);
+
+            actorProxy.JsonSerializerOptions ??= options.JsonSerializerOptions ?? this.DefaultActorProxyOptions?.JsonSerializerOptions;
```

Comment:
Will `actorProxy.JsonSerializerOptions` ever be non-null at this point?

Reply:
It would be better to pass this in to initialize

Reply:
Following your comment on passing the `JsonSerializerOptions` into the `Initialize` method I opted to pass the whole `ActorProxyOptions` instance in.